test-harness/tests/test_cache.py

```diff
@@ -668,3 +668,249 @@ def test_if_early_boot_stage_can_recover_from_a_bit_slow_backend(
             with GuardedSubprocess(v):
                 time.sleep(backend_request_timeout * 0.3 + 1)  # let it warm-up!
                 ping_mesos_agent(ar, superuser_user_header)
+
+
+class TestCacheMarathon:
+
+    def test_upstream_wrong_json(
+            self, nginx_class, mocker, superuser_user_header):
+        filter_regexp = {
+            "Cannot decode Marathon apps JSON: ": SearchCriteria(1, True),
+        }
+
+        ar = nginx_class()
+
+        # Send wrong non-json response content
+        mocker.send_command(endpoint_id='http://127.0.0.1:8080',
+                            func_name='apps_response_raw',
+                            aux_data=b"wrong response")
+
+        url = ar.make_url_from_path('/service/nginx-enabled/foo/bar/')
+        with GuardedSubprocess(ar):
+            # Register Line buffer filter:
+            lbf = LineBufferFilter(filter_regexp,
+                                   timeout=5,  # Just to give LBF enough time
+                                   line_buffer=ar.stderr_line_buffer)
+
+            # Trigger cache update by issuing request:
+            resp = requests.get(url,
+                                allow_redirects=False,
+                                headers=superuser_user_header)
+            assert "cache state is invalid" in resp.content.decode('utf-8')
+            assert resp.status_code == 503
+
+            lbf.scan_log_buffer()
+
+        assert lbf.all_found
```

Comment:
`lbf.extra_matches == {}` is enough.

`all_found` is a bit of a legacy thing - I am wondering if it shouldn't be removed. It was meant as a simplified way to check if all matching succeded, but now I see that this is confusing people a bit.